src/wasm-emscripten.cpp

```diff
@@ -137,26 +104,39 @@ void generateStackAllocFunction(LinkerObject& linker) {
   addExportedFunction(wasm, function);
 }
 
-void generateStackRestoreFunction(LinkerObject& linker) {
-  Module& wasm = linker.wasm;
-  Builder builder(wasm);
+void EmscriptenGlueLinker::generateStackRestoreFunction() {
   Name name("stackRestore");
   std::vector<NameType> params { { "0", i32 } };
   Function* function = builder.makeFunction(
     name, std::move(params), none, {}
   );
   GetLocal* getArg = builder.makeGetLocal(0, i32);
-  Store* store = generateStoreStackPointer(builder, linker, getArg);
+  Store* store = generateStoreStackPointer(getArg);
 
   function->body = store;
 
   addExportedFunction(wasm, function);
 }
 
-void generateRuntimeFunctions(LinkerObject& linker) {
-  generateStackSaveFunction(linker);
-  generateStackAllocFunction(linker);
-  generateStackRestoreFunction(linker);
+void EmscriptenGlueLinker::generateRuntimeFunctions() {
+  generateStackSaveFunction();
+  generateStackAllocFunction();
+  generateStackRestoreFunction();
+}
+
+void EmscriptenGlueLinker::generateMemoryGrowthFunction() {
```

Comment:
Perhaps for a future PR but it occurred to me that I think we actually don't need the memory growth function anymore. IIRC it existed because some Module glue wants to grow the memory sometimes, which in asm.js it just did in JS, but for wasm it has to call into the module to execute `grow_memory`. But since we wrote this, I think we just added [`WebAssembly.Memory.prototype.grow`](https://github.com/WebAssembly/design/blob/master/JS.md#webassemblymemoryprototypegrow). So I think we can remove this, and just use that in the JS glue.

Reply:
sgtm, also yeah future-PR